tests/unit/test_operator.py

```diff
@@ -422,6 +431,110 @@ def test_sidebar_relation_and_config_together(
+        # Should include both relation- and config-based items, ordered as set in config
+        actual_items = [
+            SidebarItem(**item) for item in json.loads(harness.charm._context["links"])
+        ]
+        assert actual_items == expected_sidebar_items_ordered
+
+    @pytest.mark.parametrize(
+        "sidebar_items, preferred_link_text, expected_result",
+        [
+            ([], ["some stuff"], []),  # Case where we have null input/output
+            # Case where we have empty reorder, so nothing should change
+            (
+                [
+                    SidebarItem(
+                        text="1",
+                        link="/1",
+                        type="item",
+                        icon="assessment",
+                    ),
+                    SidebarItem(
+                        text="2",
+                        link="/1",
+                        type="item",
+                        icon="assessment",
+                    ),
+                ],
+                [],
+                [
+                    SidebarItem(
+                        text="1",
+                        link="/1",
+                        type="item",
+                        icon="assessment",
+                    ),
+                    SidebarItem(
+                        text="2",
+                        link="/1",
+                        type="item",
+                        icon="assessment",
+                    ),
+                ],
+            ),
+            # Case where we have links that should be reordered
+            (
+                [
+                    SidebarItem(
+                        text="3",
+                        link="/3",
+                        type="item",
+                        icon="assessment",
+                    ),
+                    SidebarItem(
+                        text="1",
+                        link="/1",
+                        type="item",
+                        icon="assessment",
+                    ),
+                    SidebarItem(
+                        text="2",
+                        link="/2",
+                        type="item",
+                        icon="assessment",
+                    ),
+                ],
+                ("2",),
+                [
+                    SidebarItem(
+                        text="2",
+                        link="/2",
+                        type="item",
+                        icon="assessment",
+                    ),
+                    SidebarItem(
+                        text="1",
+                        link="/1",
+                        type="item",
+                        icon="assessment",
+                    ),
+                    SidebarItem(
+                        text="3",
+                        link="/3",
+                        type="item",
+                        icon="assessment",
+                    ),
+                ],
+            ),
+        ],
+    )
```

Comment:
Can you add a case where sidebar-link-order config has more than one value to make sure the order is as expected in the spec?